apis/nucleus/src/components/listbox/ListBox.jsx

```diff
@@ -105,6 +106,26 @@ export default function ListBox({
     });
   }
 
+  const updateScreenReaderSearchText = ({ inputText, listCount }) => {
+    if (inputText) {
+      let t;
+      switch (listCount) {
+        case 0:
+          t = 'Listbox.NoMatchesForYourTerms';
+          break;
+        case 1:
+          t = 'ScreenReader.OneSearchResult';
+          break;
+        default:
+          t = 'ScreenReader.ManySearchResults';
+          break;
+      }
+
+      const srText = translator.get(t, [listCount]);
+      setScreenReaderText(srText);
+    }
+  };
```

Comment:
If you want to, move some of this function outside so you can write unit tests for it:
```
const getScreenReaderSearchText = ({ inputText, listCount }) => { … }
```
```
const srText = getScreenReaderSearchText({ inputText, listCount });
setScreenReaderText(srText);
```